pkg/k8s/apps/interface.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"context"
+
+	"github.com/okteto/okteto/pkg/model"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type App interface {
```

Comment:
This is a fairly large interface. Can it be broken into smaller interfaces and possibly combined into smaller interfaces is my thought. I need to look more at the various use cases and implementations but it seems like we through all requirements into the interface and we may be able to leave some functionality out completely and be more flexible.

Reply:
Yes, I am planing to simplify it after this iteration. Once the webhook dependency is removed, many of these methods won't be needed